API page on website.

willyb321 opened the pull request:
This PR adds an API page to the website, and also fixes #76 .

PeterDaveHello commented:
Would you please remove the twitter part?

PeterDaveHello commented:
I think we should also provide some output example and links example including both human readable version and not.

willyb321 commented:
@PeterDaveHello I'll get that done now.

PeterDaveHello commented:
@willyb321 Btw, would you please the screenshot here so the others can join the discussion easily?  Thanks.

willyb321 commented:
@PeterDaveHello [Here](http://i.imgur.com/zYSNom0.png) is a screenshot of the page currently.

willyb321 commented:
Alright I reckon I've got the examples down.

PeterDaveHello commented:
it comes `You can "search" cdnjs via our API:` with the url not searching search, should adjustment this part.

PeterDaveHello commented on templates/api.html:
```diff
@@ -0,0 +1,91 @@
+		<br>
+		<pre><code>
+https://api.cdnjs.com/libraries?output=human
+https://api.cdnjs.com/libraries?search=[query]?output=human</code></pre>
```
The output indent looks strange here?

willyb321 commented:
> Can we just put some examples directly in the same page but not only the links?

Do you mean an iframe or an image or ?  

> it comes You can "search" cdnjs via our API: with the url not searching search, should adjustment this part.  

Fixed.  

> The output indent looks strange here?

Fixed  
@PeterDaveHello

PeterDaveHello commented:
@willyb321 I mean just a code block as the same output from api :)

willyb321 commented:
@PeterDaveHello ah yes didn't think of that, will do.

willyb321 commented:
Ok, code blocks for api are done. [Here](http://i.imgur.com/fhn9RzU.png) is a snippet from the page.

PeterDaveHello commented:
1. Looks like there are some unnecessary blank lines.
2. I think we can remove `(minified)` since that's what the API will do originally.
3. For the human-readable output, let's also apply hightlightjs syntax hightlight.
4. Please also display the whole url with query strings in examples without just links.

Looks like you cut some of the partial output in the codeblock, I'm not sure if this could be good for us, will think about it, we could work on the other parts first.

PeterDaveHello commented:
We should also update the route in apiServer.js to redirect api.cdnjs.com to here.

PeterDaveHello commented:
@willyb321 btw, please add (I added them for you already) `!` in the picture syntax for screenshot so that we can preview it directly, thanks.

willyb321 commented:
Ok will get that done in the morning.

PeterDaveHello commented:
@willyb321 No problem!

willyb321 commented:
@PeterDaveHello Ok, I've done 1,2,3,4. Do tell me if theres anything else you need.

willyb321 commented:
Here is a screenshot for the current page.
![image](https://cloud.githubusercontent.com/assets/9222609/20912171/c200de26-bbc1-11e6-8ff7-b086113e3323.png)

PeterDaveHello commented on templates/api.html:
```diff
@@ -1,5 +1,11 @@
+<head>
+<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/styles/default.min.css">
+<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/highlight.min.js"></script>
```
We have it in the layout.
https://github.com/cdnjs/new-website/blob/master/templates/layout.html#L163

BTW, please always use https and SRI.

willyb321 commented:
Ok will get that fixed. Just commuting atm.

willyb321 commented:
Ok I think thats what you wanted.

willyb321 commented:
@PeterDaveHello do you mean all the url's or just the ones in the example section?

PeterDaveHello commented:
@willyb321 the examples, thanks

willyb321 commented:
@PeterDaveHello ok done.

PeterDaveHello commented:
@willyb321 can we use some more precise query like https://api.cdnjs.com/libraries?search=ractive, so that we don't need to cut out the results in the codeblocks? Thanks.

willyb321 commented:
@PeterDaveHello ok, done.

PeterDaveHello commented:
@willyb321 Thanks for your super prompt reply!

PeterDaveHello commented:
@willyb321 Looks like the highlight isn't working?
BTW, in the examples, would you please move the <a> link tag from the "text" to the "url"? Thanks.

PeterDaveHello commented:
I would like to update the heading part to:

--

You can query cdnjs via our API as below

Without any query parameters it will return the name and main file URL of every library on cdnjs:
`https://api.cdnjs.com/libraries`

To search, use: 
`https://api.cdnjs.com/libraries?search=[query]`

PeterDaveHello commented:
> API will reture minified result by default, if you wanna have a human readable result, try `output=human `like so: 
->
> API will reture minified result by default, if you wanna have a human readable result, try `output=human` like so:

PeterDaveHello commented:
@PeterDaveHello For the example output been cut out, please add notes about that so that user won't feel strange :0

willyb321 commented:
@PeterDaveHello ok, done!

PeterDaveHello commented:
@willyb321 Since not all of them?

willyb321 commented:
@PeterDaveHello if you are talking about the cut out, I ended up fixing that so that I didn't need to cut out anything.

PeterDaveHello commented:
@willyb321 and the links and highlight :)

willyb321 commented:
@PeterDaveHello oh, oops!

willyb321 commented:
@PeterDaveHello Ok, I've got that done. Sorry about the slow reply, fell asleep lol.

PeterDaveHello commented:
@willyb321 no worry at all, don't need to be nervous :)

PeterDaveHello commented:
The url, fields block, should we use `<code>` there? I thought the hightlight apply not suitable style to them.

willyb321 commented:
@PeterDaveHello I reckon its fine as is.

PeterDaveHello commented:
The link part in example, actually I hope the `<a>` for link only apply to the url part inside `(`&`)`

PeterDaveHello commented:
> @willyb321 For the example output been cut out, please add notes about that so that user won't feel strange :0

This part I didn't see the update yet.

willyb321 commented:
@PeterDaveHello ok so something like this ? 
![image](https://cloud.githubusercontent.com/assets/9222609/20991900/092f3c74-bd35-11e6-96ff-223e61d34d64.png)

PeterDaveHello commented:
@willyb321 
> Specific Library Assets (https://api.cdnjs.com/libraries?search=mustcache)

willyb321 commented:
@PeterDaveHello ok thats all done. Here's an image 
![image](https://cloud.githubusercontent.com/assets/9222609/21031222/7a023142-bdf6-11e6-8eea-07cb36e32b90.png)

PeterDaveHello commented:
@willyb321 Thanks

willyb321 commented:
@PeterDaveHello ok so is that all you need?

PeterDaveHello commented:
@willyb321 the indent in human readable result is wrong.

willyb321 commented:
@PeterDaveHello ok that should be better.

PeterDaveHello commented on templates/api.html:
```diff
@@ -0,0 +1,167 @@
+<pre><code class="json">{"results":[{"name":"1140","latest":"https://cdnjs.cloudflare.com/ajax/libs/1140/2.0/1140.min.css"},{"name":"1000hz-bootstrap-validator","latest":"https://cdnjs.cloudflare.com/ajax/libs/1000hz-bootstrap-validator/0.11.6/validator.min.js"},{"name":"10up-sanitize.css","latest":"https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/4.1.0/sanitize.min.css"},{"name":"16pixel ...
+</code></pre>
+<br>
+All Libraries (Human Readable) (<a>https://api.cdnjs.com/libraries?output=human</a>) (Cut down a bit)<br>
```
this link doesn't work at all ... don't we need `href` in the `<a>` tag?

PeterDaveHello replied:
BTW, let's use a new line for the link, and remove `(`, `)`, so it'll look like:

All Libraries Human Readable output (Cut down a bit in the example)
https://api.cdnjs.com/libraries?output=human

willyb321 commented:
@PeterDaveHello Ok, that should now be fixed.

willyb321 commented:
![Here](http://i.imgur.com/C7OgIaj.png) is a pic

PeterDaveHello commented:
@willyb321 thanks, hope this is the final fix:
The example output in All Libraries part, would you please use the same content for human readable part?

BTW, would you please squash/fixup these commits into a single commit? Thanks.

willyb321 commented:
@PeterDaveHello ok its squashed now, and the example is the same.

PeterDaveHello commented:
Awesome, @willyb321 thank you for your patience and contribution!
